src/Logger.ts

```diff
@@ -0,0 +1,41 @@
+import { EventEmitter } from 'events';
+
+import Entry from './Entry';
+import TimeProvider from './TimeProvider';
+
+/**
+ * @author Maciej Chałapuk (maciej@chalapuk.pl)
+ */
+export class Logger<Level> extends EventEmitter {
```

Comment:
Why is it parametrised with `Level`?

Reply:
I didn't want to hardcode particular levels at this level of abstraction.